merkle/common_test.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package merkle
+
+import (
+	"testing"
+)
+
+func TestBit(t *testing.T) {
+	for _, tc := range []struct {
+		index []byte
+		i     int
+		want  uint
+	}{
+		{index: h2b("00"), i: 0, want: 0},
+		{index: h2b("00"), i: 7, want: 0},
+		{index: h2b("000b"), i: 0, want: 1},
+		{index: h2b("000b"), i: 1, want: 1},
+		{index: h2b("000b"), i: 2, want: 0},
+		{index: h2b("000b"), i: 3, want: 1},
```

Comment:
This doesn't fully test crossing byte boundaries. I'd add a couple of test cases.